### ISSUE TITLE
Allow variable on the command line to override playbooks

### ISSUE THREAD

**Issue by larsks** [ansible/ansible#1324]
It would be nice if variable specified on the command line would override values in playbooks.  This would, for example, allow one to distribute playbooks with some sort of default values...

```
vars:
- server: www
```

...and then let someone run the playbook _without modifying it_ by simply "filling in the blanks" on the command line:

```
ansible-playbook -e server=web.example.com playbook.yml
```

It's fairly common for things specified on the command line to take precedence...e.g., `make` will take variables, in ascending order of precedence, from the environment, from the Makefile, and finally from the command line.

This also nicely mirrors the the `include:` syntax (`include: wordpress.yml user=timmy`).

**Comment by mpdehaan** [comment]
The precedence between --extra-vars and vars is currently undefined, so I am fine with moving the precendence rules around this such that it has absolute lowest precedence 

http://ansible.cc/docs/playbooks2.html#understanding-variable-precedence

(Note that inventory variables already do clobber the command line values, it's just that vars and -e do not work in the order you'd expect)

Patches would definitely be accepted, but we're nearing our 0.8 feature freeze, so moving this into the 0.9 bucket.

**Comment by mpdehaan** [comment]
fixed per above on 0.9

**Comment by cheeming** [comment]
This was reverted at d585fe83bcadc5038e0e6ae8f2e22904a35db90f (FYI for those who are wondering about this)
